src/test/java/bio/terra/service/snapshotbuilder/utils/SearchConceptsQueryBuilderTest.java

```diff
@@ -52,7 +52,7 @@ OR CONTAINS_SUBSTR(c.concept_code, 'cancer')))
 
     String expectedAzureQuery =
         """
-            SELECT TOP 100 c.concept_name, c.concept_id, COUNT(DISTINCT o.person_id) AS count
+            SELECT TOP 100 c.concept_name, c.concept_id, COUNT(DISTINCT o.person_id) AS count, 1 AS has_children
```

Comment:
is there a way to refactor the tests so the shared parts of the generated sql are reused? and then the differences are emphasized more

Reply:
I'm of two minds; on the one hand, I don't like duplication and any duplication like this requires updating text in many places when things change. On the other hand, I don't like making our tests complicated, and adding code that conditionally builds a compare text doesn't seem good either. So for now I'm leaning toward making the tests simpler by only asserting against a complete string.

But I agree that this is probably not good in the long term. One approach would be to assert on `Query` itself instead of its generated SQL. Then the test could only assert on things that the `Query` code itself is doing. Another would be to mock any subqueries that are covered by other tests, so they're not tested here. I can make a separate ticket to look into these kinds of improvements.

Reply:
I created https://broadworkbench.atlassian.net/browse/DC-989 for this work